src/libasr/ASR.asdl

```diff
@@ -263,7 +263,8 @@ expr
     | DictConstant(expr* keys, expr* values, ttype type)
     | DictLen(expr arg, ttype type, expr? value)
     | Var(symbol v)
-    | ArrayRef(symbol v, array_index* args, ttype type, expr? value)
+    | ArrayItem(symbol v, array_index* args, ttype type, expr? value)
+    | ArraySection(symbol v, array_index* args, ttype type, expr? value)
```

Comment:
```suggestion
    | ArraySection(expr v, array_index* args, ttype type, expr? value)
```

Reply:
I wouldn't prefer to do this change in this PR. Fixing `ArrayItem` is the next stop in my priority list. So, I would focus on it once this gets in.